tests/Microsoft.Identity.Web.Test/L1L2CacheTests.cs

```diff
@@ -149,6 +151,7 @@ private async Task CreateL1L2TestWithAbsoluteOptions(double time)
 
             // Assert
             Assert.Equal(memoryCacheExpectedCount, _testCacheAdapter._memoryCache.Count);
+            await Task.Delay(10).ConfigureAwait(false); // needed for L2 fire&forget
```

Comment:
This is not a reliable way to test. Typically what happens is that you'll get failures on the CI build every now and then and have increase the delay. As we add more tests, this logic gets copied and the timer gets 

To test "fire and forget" in this case it would be simplest to inject a custom IDistributedCache which internally uses a semaphore. Then the test can wait on the semaphore and `IDistributedCache.Set` will release the sempahore.

Here's an test that does this with ManualResetEvent instead of semaphore: https://github.com/AzureAD/microsoft-authentication-extensions-for-dotnet/blob/master/tests/Microsoft.Identity.Client.Extensions.Msal.UnitTests/MsalCacheHelperTests.cs#L174

Reply:
Yes, I know it's not a good way to test. That's why I had it marked as a draft until I had time to fix the tests. Thanks for the advice. Have fixed.